src/routes/api/users.js

```diff
@@ -1,82 +1,25 @@
 import { Router } from 'express';
 import passport from 'passport';
+import userController from '../../controllers/userController';
 
 const router = Router();
 
-router.get('/user', (req, res, next) => {
-  User.findById(req.payload.id)
-    .then((user) => {
-      if (!user) {
-        return res.sendStatus(401);
-      }
-      return res.json({ user: user.toAuthJSON() });
-    })
-    .catch(next);
-});
+const { forgotPassword, resetPassword } = userController;
 
-router.put('/user', (req, res, next) => {
-  User.findById(req.payload.id)
-    .then((user) => {
-      if (!user) {
-        return res.sendStatus(401);
-      }
+router.get('/user');
 
-      // only update fields that were actually passed...
-      if (typeof req.body.user.username !== 'undefined') {
-        user.username = req.body.user.username;
-      }
-      if (typeof req.body.user.email !== 'undefined') {
-        user.email = req.body.user.email;
-      }
-      if (typeof req.body.user.bio !== 'undefined') {
-        user.bio = req.body.user.bio;
-      }
-      if (typeof req.body.user.image !== 'undefined') {
-        user.image = req.body.user.image;
-      }
-      if (typeof req.body.user.password !== 'undefined') {
-        user.setPassword(req.body.user.password);
-      }
+router.put('/user');
 
-      return user.save().then(() => res.json({ user: user.toAuthJSON() }));
-    })
-    .catch(next);
-});
+router.post('/users');
 
-router.post('/users/login', (req, res, next) => {
-  if (!req.body.user.email) {
-    return res.status(422).json({ errors: { email: "can't be blank" } });
-  }
+// @route POST /api/v1/users/forgotpassword
+// @desc Generate User Password Reset / Returning JWT Token
+// @access Public
+router.post('/forgotpassword', forgotPassword);
 
-  if (!req.body.user.password) {
-    return res.status(422).json({ errors: { password: "can't be blank" } });
-  }
-  passport.authenticate('local', { session: false }, (
-    err,
-    user,
-    info
-  ) => {
-    if (err) {
-      return next(err);
-    }
-
-    if (user) {
-      return res.json({ user: user.toAuthJSON() });
-    }
-    return res.status(422).json(info);
-  })(req, res, next);
-});
-
-router.post('/users', (req, res, next) => {
-  const user = new User();
-
-  user.username = req.body.user.username;
-  user.email = req.body.user.email;
-  user.setPassword(req.body.user.password);
-
-  user.save()
-    .then(() => res.json({ user: user.toAuthJSON() }))
-    .catch(next);
-});
+// @route POST /api/v1/users/resetpassword/:id/
+// @desc Resets a User Password / Returns a new Password
+// @access Public
+router.post('/resetpassword/:id', resetPassword);
```

Comment:
@conquext  you didn't version your API.
@marusoft @wahaaabello @dewaleolaoye @Dolapo47 you reviewed this.

Reply:
@rovilay api is versioned. pls see src/routes/api/index.js